### PR TITLE
Fixes #22 - Update the app to use Collection Membership branch of CC

### LINKED ISSUE #22: Review Collection Membership branch of CC
Per https://groups.google.com/forum/#!topic/hydra-tech/7_2jSVL8CNk

This would probably be important for us, in practice. It sounds like this branch will get merged eventually. Esme is guessing 2.0 given the breaking changes and keeping with semantic versioning, that makes sense. We may, if we start doing larger migration tests, want to try out using this branch. It would also allow us to provide feedback based on our experience.

### PR DESCRIPTION
**Pull request by VivianChu**

### CODE REVIEW

**Comment by VivianChu** [comment]
@ucsdlib/developers - Please review, comment, etc... Thanks

File: app/services/indexes_attributes.rb
Comment (mcritchlow) [inline app/services/indexes_attributes.rb]: Could the Enumerable `find` method be used here instead?

Comment (VivianChu) [reply app/services/indexes_attributes.rb]: @mcritchlow Yes, I think it could be used here.  However, I moved the original code that @lsitu wrote from the WorkIndexer https://github.com/ucsdlib/dams5-cc-pilot/blob/master/app/indexers/work_indexer.rb to here so it could be reused by both CollectionIndexer and WorkIndexer.  If @lsitu doesn't mind that I update his original code, then I will update it to use Enumerable and test it.

Comment (lsitu) [reply app/services/indexes_attributes.rb]: @VivianChu Please feel free to make the changes. Thanks.

**Comment by VivianChu** [comment]
@mcritchlow - The code has been updated. Could you review again?  Thanks

File: app/services/indexes_attributes.rb
Comment (mcritchlow) [inline app/services/indexes_attributes.rb]: Looks good. This could, I think, also be made into a one-liner with a con of potentially being a bit harder to read. With that in mind, here's that one-liner approach that you can consider. Overall, I'm 👍 on this PR.

As for the one-liner, I believe you can pass a proc to `find` in the case where there isn't a match. So, you might be able to do something like:

```
Array(cv_list.find(proc { [uri] }) { |pair| pair[1] == uri }).first
```

**Comment by VivianChu** [comment]
@mcritchlow I tested your one line code and it worked fine.  I updated the code to use it.

**Comment by lsitu** [comment]
@mcritchlow @VivianChu 👍 
With so many big changes on going at the CC side, I am wondering whether we should keep it a separate branch for now to avoid any potential issues with keeping CC up-to-date for the collection membership branch. Thoughts?

**Comment by mcritchlow** [comment]
@lsitu @VivianChu That's a very good point, though this repo is still a pilot and not necessarily slated for production any time soon anyway. It's hard to imagine using a version of CC on our end that doesn't use the collection membership solution in the branch.

That said, I wonder @VivianChu might be able to check in with Esme to see if he knows any more about the future of this branch. Perhaps that might guide our thinking about what to do for now?

**Comment by VivianChu** [comment]
@mcritchlow @lsitu Sounds good.  I will check in with Esme.

**Comment by lsitu** [comment]
@mcritchlow @VivianChu I am fine to merge it now and update it to the latest CC version once the collection-membership branch is merged to CC.

**Comment by mcritchlow** [comment]
@lsitu I agree I think that might be the best path. Can we get a Review from someone approving this PR then?

**Review by hweng — COMMENTED** [review]
:+1:

**Review by lsitu — COMMENTED** [review]
Approved it since we all agree to merge it for now.

**Review by lsitu — APPROVED** [review]

**Pull request merged**
